testing/dart/color_test.dart

```diff
@@ -27,23 +27,27 @@ void main() {
   });
 
   test('color created with out of bounds value', () {
+    Object? exception;
     try {
       const Color c = Color(0x100 << 24);
       final Paint p = Paint();
       p.color = c;
     } catch (e) {
-      expect(e != null, equals(true));
+      exception = e;
     }
+    expect(exception != null, equals(true));
   });
 
   test('color created with wildly out of bounds value', () {
+    Object? exception;
     try {
       const Color c = Color(1 << 1000000);
       final Paint p = Paint();
       p.color = c;
     } catch (e) {
-      expect(e != null, equals(true));
+      exception = e;
     }
+    expect(exception != null, equals(true));
```

Comment:
I thought the original intend of these two tests was that there would be an exception, but rewriting the tests to actually assert that (as shown here) makes the test fail. Needs some more investigation...

Reply:
Interestingly, on the web SDK side these tests are explicitly testing that you can create these with out-of-bounds values: 

https://github.com/flutter/engine/blob/bffb98352446ae620ea40dbef0e38562b24bde1e/lib/web_ui/test/ui/color_test.dart#L37-L47

Maybe that is the intend here after all?

Reply:
Since the color constructor is just removing anything above `0xFFFFFFFF` I am going to assume that creating a color with out-of-bounds values should be legal and not throw. I'll adjust these tests accordingly. 

https://github.com/flutter/engine/blob/bffb98352446ae620ea40dbef0e38562b24bde1e/lib/ui/painting.dart#L108